template/cordova/lib/PluginInfo.js

```diff
@@ -0,0 +1,150 @@
+var semver = require('semver');
+var CommonPluginInfo = require('cordova-common').PluginInfo;
+
+var MANIFESTS = {
+    'windows': {
+        '8.1.0': 'package.windows.appxmanifest',
+        '10.0.0': 'package.windows10.appxmanifest'
+    },
+    'phone': {
+        '8.1.0': 'package.phone.appxmanifest',
+        '10.0.0': 'package.windows10.appxmanifest'
+    },
+    'all': {
+        '8.1.0': ['package.windows.appxmanifest', 'package.phone.appxmanifest'],
+        '10.0.0': 'package.windows10.appxmanifest'
+    }
+};
+
+var SUBSTS = ['package.phone.appxmanifest', 'package.windows.appxmanifest', 'package.windows10.appxmanifest'];
+var TARGETS = ['windows', 'phone', 'all'];
+/*
+A class for holidng the information currently stored in plugin.xml
+It's inherited from cordova-common's PluginInfo class
+In addition it overrides getConfigFiles method to use windows-specific logic
+ */
+function PluginInfo(dirname) {
+    // We needn't use 'util.inherit' in this case because there aren't any methods on prototype in  cordova-common's PluginInfo class
+    CommonPluginInfo.apply(this, arguments);
+    var parentGetConfigFiles = this.getConfigFiles;
+    var parentGetEditConfigs = this.getEditConfigs;
+
+    this.getEditConfigs = function(platform) {
+        var editConfigs = parentGetEditConfigs(platform);
+        return processChanges(editConfigs);
+    };
+
+    this.getConfigFiles = function(platform) {
+        var configFiles = parentGetConfigFiles(platform);
+        return processChanges(configFiles);
+    };
+
+    function processChanges(changes) {
+        var hasManifestChanges  = checkManifestChanges(changes);
+
+        // Demux 'package.appxmanifest' into relevant platform-specific appx manifests.
+        // Only spend the cycles if there are version-specific plugin settings
+        if (hasManifestChanges)
+        {
+            var oldChanges = changes;
+            changes = [];
+
+            oldChanges.forEach(function(change, changeIndex) {
+                // Only support semver/device-target demux for package.appxmanifest
+                // Pass through in case something downstream wants to use it
+                if (change.target !== 'package.appxmanifest') {
+                    changes.push(change);
+                    return;
+                }
+
+                // No semver/device-target for this config-file, pass it through
+                if (!(hasChangeVersion(change) || hasChangeTarget(change))) {
+                    // New windows template separate manifest files for Windows10, Windows8.1 and WP8.1
+                    changes = changes.concat(demuxChangeWithSubsts(change));
+                    return;
+                }
+
+                var targetDeviceSet = getDeviceTargetSetForChange(change);
+                changes = replaceChangeTarget(targetDeviceSet, change, changes);
```

Comment:
Consider renaming `replaceChangeTarget` to something more appropriate as it doesn't really replace anything in the original array